config/settings/base.py

```diff
@@ -28,7 +28,10 @@
 # https://docs.djangoproject.com/en/dev/ref/settings/#language-code
 LANGUAGE_CODE = "fr-fr"
 # https://docs.djangoproject.com/en/dev/ref/settings/#site-id
-SITE_ID = 1
+SITE_ID = env(
```

Comment:
Ça me semble un peu problématique, parce que le prérequis est qu'on utilise un seul environnement Scalingo.

Il me semble que ce serait plus approprié de détecter le site en se basant sur le domaine, comme indiqué dans la doc de Django :

The [SITE_ID](https://docs.djangoproject.com/en/5.0/ref/settings/#std-setting-SITE_ID) setting specifies the database ID of the [Site](https://docs.djangoproject.com/en/5.0/ref/contrib/sites/#django.contrib.sites.models.Site) object associated with that particular settings file. If the setting is omitted, the [get_current_site()](https://docs.djangoproject.com/en/5.0/ref/contrib/sites/#django.contrib.sites.shortcuts.get_current_site) function will try to get the current site by comparing the [domain](https://docs.djangoproject.com/en/5.0/ref/contrib/sites/#django.contrib.sites.models.Site.domain) with the host name from the [request.get_host()](https://docs.djangoproject.com/en/5.0/ref/request-response/#django.http.HttpRequest.get_host) method.

Tu avais une raison particulière de te baser sur l'environnement ?

Reply:
J'ai eu exactement le même raisonnement. Un seul hebergement et on descrimine à partir du nom de domaine (je m'etais créé une redirection "haie.local" sur mon environnement de dev pour l'occasion).

Les choses se sont compliquées lorsque j'ai voulue définir un model manager pour l'objet regulation. [Il y en a un proposé par django](https://docs.djangoproject.com/en/5.0/ref/contrib/sites/#the-currentsitemanager), mais voici ce que dit la doc:

> Mandatory [SITE_ID](https://docs.djangoproject.com/en/5.0/ref/settings/#std-setting-SITE_ID)
> The CurrentSiteManager is only usable when the [SITE_ID](https://docs.djangoproject.com/en/5.0/ref/settings/#std-setting-SITE_ID) setting is defined in your settings.

On peut surement contourner ce probleme "la plupart du temps" en trainant la requete (ou au moins le domaine) jusqu'au manager. Mais il existe des cas ou il n'y a pas de requete, et donc pas moyen de connaitre le site : e.g. task, migration, command...

Il y a également [cette partie de la doc](https://docs.djangoproject.com/en/5.0/ref/contrib/sites/#how-django-uses-the-sites-framework) qui me chagrine un peu. Le "sites framework" semble utilisé par django, mais je ne sais pas s'il se base sur la requete ou sur le SITE_ID des settings.

Reply:
Mmm… Si l'utilisation du `CurrentSiteManager` qui n'économise qu'un seul `filter` dans une seule `queryset` remet en cause toute notre architecture d'hébergement, peut-être que c'est plutôt le `CurrentSiteManager` qui devrait être remplacé ?

Tu as raison sur le fait que la Moulinette n'a pas accès à la request, donc pas accès au « current site ». Pour le coup, je ne suis pas convaincu que ce soit le mécanisme qu'il faille utiliser pour assigner des réglementations à des sites différents, mais peut-être que ça ne fait pas partie de cette PR.